src/set_theory/cardinal/ordinal.lean

```diff
@@ -805,6 +812,15 @@ lemma mk_finsupp_of_infinite (α β : Type u) [infinite α] [has_zero β]
   [nontrivial β] : #(α →₀ β) = max (#α) (#β) :=
 by simp
 
+@[simp] lemma mk_finsupp_nat (α : Type u) [nonempty α] : #(α →₀ ℕ) = max (#α) ℵ₀ :=
```

Comment:
You should be able to more generally prove `#(α →₀ β) = max (#α) (#β)` when `α` is nonempty and `β` is infinite. The proof proceeds by casing on whether `α` is finite or infinite and using the existing theorems on the cardinality of finitely supported functions.